src/applications/static-pages/subscribeComponentAnalyticsEvents.js

```diff
@@ -0,0 +1,50 @@
+import recordEvent from 'platform/monitoring/record-event';
+
+const analyticsEvents = [
+  {
+    componentName: 'Modal',
+    prefix: 'modal',
+    actions: [{ action: 'show', name: 'int-modal-click' }],
+  },
+  {
+    componentName: 'AdditionalInfo',
+    prefix: 'additional-info',
+    actions: [
+      { action: 'expand', name: 'int-additional-info-expand' },
+      { action: 'collapse', name: 'int-additional-info-collapse' },
+    ],
+  },
+];
```

Comment:
If the `prefix` can be removed, this might work as a data structure like so:

```js
const analyticsEvents = {
    'Modal': [ { action: 'expand', name: 'int-modal-click' } ],
    'AdditionalInfo': [
        { action: 'expand', name: 'int-additional-info-expand' },
        { action: 'collapse', name: 'int-additional-info-collapse' },
    ],
}
```

If you wanted to simplify the structure and offload some of the complexity into the logic, you could even remove the `int-` prefix from the name and be sure to add it to the `dataLayer` object before making the `recordEvent` call. If the event names can be matched appropriately (between what we are firing from `component-library` and what we are expecting in GA), you can also construct `action.name` based on concatenating the key ("Modal"), and a portion of the value.

I'm not on the team though, so this is just me speculating. I don't know if having a simpler data structure with more transforming going on is desirable here or not :shrug: .

Reply:
Ah - unfortunately I don't think we can automate it that much.  For instance - the _AlertBox_ link clicks will be prefixed with `nav-` and not `int-` since the GTM tagging classifies them as navigation and not interaction ( ¯\\\_(ツ)\_/¯ ) 

I do like your simplified lookup structure, so I switched over to that.

And dropped the _prefix_ param for now in lieu of the lodash solution.